feat: 增加webrtc代理拉流

baigao-X opened the pull request:
- 增加WHEP代理拉流
- 增加WebRtc p2p 模式
  - 增加 ICE FULL模式
  - 增加STUN/TURN 服务器
  - 增加websocket 信令
  - 增加P2P代理拉流

baigao-X committed:
feat: 增加webrtc代理拉流

- 增加WHEP代理拉流
- 增加WebRtc p2p 模式
  - 增加 ICE FULL模式
  - 增加STUN/TURN 服务器
  - 增加websocket 信令
  - 增加P2P代理拉流

xia-chu committed:
Merge branch 'draft-webrtc-p2p' of github.com:baigao-X/ZLMediaKit

xia-chu committed:
修复编译问题

xia-chu committed:
添加bom头

xia-chu committed:
tab替换为空格键

xia-chu committed:
修复win编译问题

xia-chu committed:
优化代码

xia-chu committed:
移动稳定，确保openssl未开启时编译通过

mtdxc committed:
规范化offerSdp:
- 不生成错误的candidate
- 生成视频的rtx ftmp
- 增加H265和AV1的ftmp

xia-chu committed:
todo

xia-chu committed:
delete return

xia-chu committed:
remove ';'

xia-chu committed:
move WebRtcArgsImp

xia-chu committed:
Merge branch 'master' of github.com:ZLMediaKit/ZLMediaKit into pr

# Conflicts:
#	webrtc/Sdp.cpp

xia-chu committed:
添加bom

baigao-X committed:
Merge branch 'ZLMediaKit:master' into draft-webrtc-p2p

baigao-X committed:
fix: 修改whip推流失败的问题

baigao-X committed:
feat: 优化部分接口的传参(未完全完成)

baigao-X committed:
feat: 优化部分接口的传参

xia-chu committed:
fix func args type

xia-chu committed:
fix typo

xia-chu committed:
fix std::move

xia-chu committed:
bug fix: pair is null

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ZLMediaKit/ZLMediaKit?pullRequest=4389) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/ZLMediaKit/ZLMediaKit?pullRequest=4389) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: baigao-X<br/>:white_check_mark: xia-chu<br/>:x: cqm<br/><hr/>**cqm** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ZLMediaKit/ZLMediaKit?pullRequest=4389) it.</sub>

baigao-X committed:
fix: webrtctransport pair is null

xia-chu committed:
Merge remote-tracking branch 'open_src/master' into pr

xia-chu committed:
update

xia-chu committed:
update 2

xia-chu committed:
update 3

xia-chu committed:
update 4

xia-chu committed:
fix build issue

xia-chu committed:
update 5

baigao-X committed:
fix bug: 修复pair send target异常问题

baigao-X committed:
fix bug: 修改Allocate Request 认证异常问题

xia-chu committed:
update 6

baigao-X committed:
feat: 增加webrtc 相关postman api

mtdxc committed:
完善指定发送目标地址相关逻辑

mtdxc committed:
remove unused include

mtdxc committed:
支持signalingSsl和IceTcp

mtdxc committed:
优化webrtc推流服务端与拉流客户端

mtdxc committed:
StunPacket代码改进:
- 增加getAttribute<T>模板，直接通过类型返回对象
- 构造增加transId参数，避免重复生成transId
- StunAttrXorPeerAddress::getAddrString -> getIp

mtdxc committed:
set default port

mtdxc committed:
WebRtcSignalingPeer支持ssl

Commit:
add missing bom header

mtdxc committed:
fixed getTimeOutSec error

fixed typo

xia-chu committed:
Merge remote-tracking branch 'origin/master' into pr

# Conflicts:
#	server/WebApi.cpp

xia-chu committed:
格式化代码

xia-chu committed:
优化参数传参

xia-chu committed:
优化代码

xia-chu committed:
优化代码

xia-chu committed:
优化代码

xia-chu committed:
优化命名空间

xia-chu committed:
修改readme

xia-chu committed:
修复编译问题

xia-chu committed:
fix bug

xia-chu committed:
remove log

xia-chu committed:
fix bug 2

xia-chu committed:
fix crash

mtdxc committed:
fixed crash OnDtlsTransportFailed

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ZLMediaKit/ZLMediaKit?pullRequest=4389) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/ZLMediaKit/ZLMediaKit?pullRequest=4389) before we can accept your contribution.<br/>**2** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: mtdxc<br/>:white_check_mark: xia-chu<br/>:x: cqm<br/>:x: baigao-X<br/><hr/>**cqm** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ZLMediaKit/ZLMediaKit?pullRequest=4389) it.</sub>

mtdxc committed:
IceTransport功能改进
- 增加dumpString和addrToStr方法改进日志
- stun请求重试数改成可配置

IceAgent功能改进
- rename _checklist -> _check_list
- gatheringCandidate 忽略INETADDR_ANY和local打头地址
- connectivityCheck _host_sockets 增加套接口类型判断

mtdxc committed:
修改onIceTransportGatheringCandidate的签名

mtdxc committed:
small change

baigao-X committed:
fix: fix some function param error

baigao-X committed:
fix: 修改拉流失败不返回结果的问题

baigao-X committed:
fix: 修改作为whip服务端不支持连接迁移的问题

baigao-X committed:
fix: 修改onCompelted 后BINGDING REQUEST 不响应问题(修复obs whip推流心跳超时问题)

xia-chu committed:
Merge remote-tracking branch 'open_src/master' into pr

# Conflicts:
#	3rdpart/ZLToolKit
#	server/WebApi.cpp
#	webrtc/WebRtcPusher.cpp

xia-chu committed:
完善webrtc推流客户端

xia-chu committed:
添加bom头

baigao-X committed:
feat: 信令增加sender字段

baigao-X committed:
feat:  candidate增加 base addr信息

baigao-X committed:
fix: 修改ipv6 v4mpped 地址 relay失败的问题

baigao-X committed:
fix: 修改部分webrtcProxyPlayerInfo 信息

baigao-X committed:
fit: 修改nominated状态,以解决一次ice过程中频繁切换candidate_pair 问题

FIXME: 需要处理priority

baigao-X committed:
fix: 增加rtc.interfaces配置,用于指定网卡，解决公网服务器使用弹性公网IP导致无法bind的问题

xia-chu committed:
版本号改成9.0

xia-chu committed:
Merge remote-tracking branch 'pr/draft-webrtc-p2p'

xia-chu committed:
更新readme

xia-chu committed:
更新macos ci
